pkg/uhttp/gocache.go

```diff
@@ -0,0 +1,129 @@
+package uhttp
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"net/http/httputil"
+	"time"
+
+	bigcache "github.com/allegro/bigcache/v3"
+)
+
+type GoCache struct {
+	ttl         time.Duration
+	rootLibrary *bigcache.BigCache
+}
+
+func NewGoCache(ctx context.Context, ttl int32) (GoCache, error) {
+	eviction := time.Duration(ttl) * time.Minute
+	c, err := bigcache.New(ctx, bigcache.DefaultConfig(eviction))
+	if err != nil {
+		return GoCache{}, err
+	}
+
+	gc := GoCache{
+		ttl:         eviction,
+		rootLibrary: c,
+	}
+
+	return gc, nil
+}
+
+func GetCacheKey(req *http.Request) string {
+	return req.URL.String()
```

Comment:
I think we should include some http headers in the cache key. For example, hitting the same URL with `Accept: application/json` and `Accept: application/xml` will return very different responses.

The headers I can think of off the top of my head:

- [Accept](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept)
- [Authorization](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization) (because using different credentials probably means we'll get different results)
- [Cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cookie)
- [Range](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range)

Reply:
Fixed